modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -1204,12 +1232,32 @@ public void onFailure(Exception e) {
      * is different to what's already in the error store.
      * This allows us to not spam the logs and only log new errors when we're about to record them in the store.
      */
-    static void recordAndLogError(String targetIndex, DataStreamLifecycleErrorStore errorStore, Exception e, String logMessage) {
-        String previousError = errorStore.recordError(targetIndex, e);
-        if (previousError == null || previousError.equals(errorStore.getError(targetIndex)) == false) {
+    static void recordAndLogError(
+        String targetIndex,
+        DataStreamLifecycleErrorStore errorStore,
+        Exception e,
+        String logMessage,
+        int signallingErrorRetryThreshold
+    ) {
+        ErrorEntry previousError = errorStore.recordError(targetIndex, e);
+        ErrorEntry currentError = errorStore.getError(targetIndex);
+        if (previousError == null || (currentError != null && previousError.error().equals(currentError.error()) == false)) {
             logger.error(logMessage, e);
         } else {
-            logger.trace(logMessage, e);
+            if (currentError != null && currentError.retryCount() % signallingErrorRetryThreshold == 0) {
+                logger.error(
+                    String.format(
+                        Locale.ROOT,
+                        "%s\nFailing since [%d], operation retried [%d] times",
+                        logMessage,
+                        currentError.firstOccurrenceTimestamp(),
+                        currentError.retryCount()
+                    ),
+                    e
+                );
+            } else {
+                logger.trace(logMessage, e);
```

Comment:
Can this be changed to include the extra `Failing since [%d], operation retried [%d] times` lines, so that if we have to turn on trace logging, we can get that info?